packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -39,10 +41,41 @@ export class Snapshot {
     await this.client.resourceSnapshot.delete(this.model.id);
   }
 
-  private get snapshotId() {
+  public requiresSynchronization() {
+    // TODO: backend should provide a specific result code for snapshots that are out of sync with the target org.
+    // Waiting for the JIRA number...
+    return (
+      this.latestReport.resultCode ===
+      ResourceSnapshotsReportResultCode.ResourcesInError
+    );
+  }
+
+  public saveDetailedReport(projectPath: string) {
+    const pathToReport = join(
+      projectPath,
+      'snapshot-reports',
+      `${this.latestReport.id}.json`
+    );
+    ensureFileSync(pathToReport);
+    writeJsonSync(pathToReport, this.latestReport, {spaces: 2});
+    return pathToReport;
+  }
+
+  public get latestReport(): ResourceSnapshotsReportModel {
+    if (this.model.reports === undefined || this.model.reports.length === 0) {
```

Comment:
Given what's on line 68:
```suggestion
    if (!Array.isArray(this.model.reports) || this.model.reports.length === 0) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray